src/particles/diagnostics/DiagnosticOutput.cpp

```diff
@@ -53,43 +54,51 @@ namespace impactx::diagnostics
         int const nLevel = tmp.finestLevel();
         for (int lev = 0; lev <= nLevel; ++lev) {
             // loop over all particle boxes
-            using ParIt = typename decltype(tmp)::ParConstIterType;
+            using ParIt = amrex::ParIterSoA<RealSoA::nattribs, IntSoA::nattribs>;
+            using ParticleType = amrex::SoAParticle<RealSoA::nattribs,IntSoA::nattribs>;
+            using ParticleTileDataType = amrex::ParticleTileData<ParticleType,RealSoA::nattribs,IntSoA::nattribs>;
             for (ParIt pti(tmp, lev); pti.isValid(); ++pti) {
                 const int np = pti.numParticles();
 
                 // preparing access to particle data: AoS
-                using PType = ImpactXParticleContainer::ParticleType;
-                auto const &aos = pti.GetArrayOfStructs();
-                PType const *const AMREX_RESTRICT aos_ptr = aos().dataPtr();
+                //using PType = ImpactXParticleContainer::ParticleType;
+                //auto const &aos = pti.GetArrayOfStructs();
+                //PType const *const AMREX_RESTRICT aos_ptr = aos().dataPtr();
 
                 // preparing access to particle data: SoA of Reals
-                auto &soa_real = pti.GetStructOfArrays().GetRealData();
-                amrex::ParticleReal const *const AMREX_RESTRICT part_px = soa_real[RealSoA::ux].dataPtr();
-                amrex::ParticleReal const *const AMREX_RESTRICT part_py = soa_real[RealSoA::uy].dataPtr();
-                amrex::ParticleReal const *const AMREX_RESTRICT part_pt = soa_real[RealSoA::pt].dataPtr();
+                //auto &soa_real = pti.GetStructOfArrays().GetRealData();
+                //amrex::ParticleReal const *const AMREX_RESTRICT part_px = soa_real[RealSoA::ux].dataPtr();
```

Comment:
@Thierry992 please do not comment the momentum components out, we need them here :)